inputPageCode.php

```diff
@@ -0,0 +1,48 @@
+<?php
+require_once('session.php');
+require_once('functions.php');
+
+$db = linkDB();
+$mice = getDataFromDB($db);
+
+if (!empty($_POST['mouseName']) && !empty($_POST['mouseBrand']) && !empty($_POST['mouseWeight']) && !empty($_POST['wirelessInput'])) {
+    $mouseName = $_POST['mouseName'];
+    $mouseBrand = $_POST['mouseBrand'];
+    $mouseWeight = $_POST['mouseWeight'];
+    if ($_POST['wirelessInput'] == 'yes') {
+        $isWireless = 1;
+    } else {
+        $isWireless = 0;
+    }
+    $mouseName = formatName($mouseName);
+    $mouseBrand = formatName($mouseBrand);
+} else {
+    header('Location: ./inputPage.php?error=missing');
+    exit;
+}
+
+if (validateInfo($_POST['mouseName'], $_POST['mouseBrand'], $_POST['mouseWeight']) !== 'true') {
```

Comment:
because your function returns a string you have now had to run the same function twice. thats not DRY and is slower than necessary

Reply:
:/ i thought it was quite elegeant to have one function return multiple potential errors and then I only need to use header once.